pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -84,18 +85,30 @@ func Add(ctx context.Context, cfg *rest.Config, namespace string) error {
 		return errors.Wrap(err, "could not create metrics Service")
 	}
 
-	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
-	// necessary to configure Prometheus to scrape metrics from this operator.
-	services := []*v1.Service{service}
-	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+	// Create a monitoring client
+	mclient, err := monclient.NewForConfig(cfg)
 	if err != nil {
-		log.Error(err, "could not create ServiceMonitor object")
-		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
-		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
-		if err == metrics.ErrServiceMonitorNotPresent {
-			metricsEnabled = false
-			return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
+		return errors.Wrap(err, "could not create monitoring client")
+	}
 
+	// If ServiceMonitor object does not exist, create one
+	// else use existing ServiceMonitor object
+	_, err = mclient.ServiceMonitors(namespace).Get(context.TODO(), windowsMetricsEndpoints, metav1.GetOptions{})
```

Comment:
You cant just check if the service monitor exists, we need to ensure that it exists with the correct spec. Consider the case where the service monitor is created by a previous version of WMCO. The expected value of the service monitor back then might differ than the current version. In that case metrics may exhibit unexpected behavior.

Reply:
Got it. So instead could I do something like "If ServiceMonitor object does not exist, create one. Else delete current one/create a new one"?

Reply:
Yep it could be handled that way